### PR TITLE
Split off bevy_image from bevy_render

### PR DESCRIPTION
**Pull request by atlv24**
# Objective

- bevy_render is gargantuan

## Solution

- Split off bevy_image

## Testing

- Ran some examples

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
Adding this to the 0.15 milestone. I don't want this to wait around because of the merge conflict pain, and merging this just after a release will make point release fixes brutal.

File: crates/bevy_image/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+] }
+bevy_utils = { path = "../bevy_utils", version = "0.15.0-dev" }
+
+# rendering
+image = { version = "0.25.2", default-features = false }
+
+# misc
+bitflags = { version = "2.3", features = ["serde"] }
+bytemuck = { version = "1.5" }
+wgpu = { version = "22", default-features = false }
```
Comment (JMS55) [inline crates/bevy_image/Cargo.toml:44]: Do you need the full wgpu here, or will wgpu-types work?

**Review by JMS55 — APPROVED** [review]

**Review by IceSentry — APPROVED** [review]

**Comment by mockersf** [comment]
I think I would prefer a more general crate than bevy_image, to be able to move other kinds of assets currently living in bevy_render like `Mesh`, rather than multiplying small crates like `bevy_image`, `bevy_mesh`, ...
maybe a `bevy_render_assets`?

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
I'm neutral on the two approaches, but I'd like to decide relatively quickly. I do like that the current split means 2D games don't have to pull in 3D mesh formats, but that can be feature-flagged within this crate.

**Comment by alice-i-cecile** [comment]
There was [more discussion on Discord](https://discord.com/channels/691052431525675048/692572690833473578/1291855507187040377), but we're moving forward with the `bevy_image` + `bevy_mesh` approach.

**Pull request merged**